cou/utils/juju_utils.py

```diff
@@ -515,3 +522,110 @@ def _check_time(self) -> None:
             raise TimeoutException(
                 f"Model {self.model_name} has not stabilized after {self.timeout} seconds."
             )
+
+
+def retry(
+    function: Optional[Callable] = None,
+    timeout: int = 60,
+    no_retry_exception: Optional[tuple] = None,
+) -> Callable:
+    """Retry function for usage in COUModel.
+
+    :param function: function to wrapped
+    :type function: Optional[Callable]
+    :param timeout: timeout in seconds
+    :type timeout: int
+    :param no_retry_exception: tuple of exception on which function will not be retried
+    :type no_retry_exception: Optional[tuple]
+    :return: wrapped function
+    :rtype: Callable
+    """
+    ignored_exceptions = (TimeoutException, *(no_retry_exception or ()))
+
+    def _wrapper(func: Callable) -> Callable:
+        @wraps(func)
+        async def wrapper(self: "COUModel", *args: Any, **kwargs: Any) -> Any:
+            start_time = datetime.now()
+            while (datetime.now() - start_time).seconds <= timeout:
+                try:
+                    return await func(self, *args, **kwargs)
+                except ignored_exceptions:
+                    # raising exception if no_retry_exception happen or TimeoutException
+                    raise
+                except Exception:
+                    logger.debug("function %s failed", func.__name__, exc_info=True)
```

Comment:
We should add some wait here. If we fail too quick this may cause problems(too much hanging connections, max number of open files etc.)